packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -783,42 +867,74 @@ abstract class RenderViewportBase<ParentDataClass extends ContainerParentDataMix
   Iterable<RenderSliver> get childrenInHitTestOrder;
 
   @override
-  void showOnScreen([RenderObject child]) {
-    RenderViewportBase.showInViewport(child: child, viewport: this, offset: offset);
-    // Make sure the viewport itself is on screen.
-    super.showOnScreen();
+  void showOnScreen({
+    RenderObject descendant,
+    Rect rect,
+    Duration duration: Duration.zero,
+    Curve curve: Curves.ease,
+  }) {
+    final Rect newRect = RenderViewportBase.showInViewport(
+      descendant: descendant,
+      viewport: this,
+      offset: offset,
+      rect: rect,
+      duration: duration,
+      curve: curve,
+    );
+    super.showOnScreen(
+      // Omitting `descendant` to get this viewport on screen.
+      rect: newRect,
+      duration: duration,
+      curve: curve,
+    );
   }
 
-  /// Make the given `child` of the given `viewport` fully visible in the
-  /// `viewport` by manipulating the provided [ViewportOffset] `offset`.
+  /// Make (a portions of) the given `descendant` of the given `viewport` fully
+  /// visible in the `viewport` by manipulating the provided [ViewportOffset]
+  /// `offset`.
+  ///
+  /// The optional `rect` parameter describes which area of the `descendant`
+  /// should be shown in the viewport. If `rect` is null, the entire
+  /// `descendant` will be revealed. The `rect` parameter is interpreted
+  /// relative to the coordinate system of `descendant`.
+  ///
+  /// The returned [Rect] describes the new location of `descendant` or `rect`
+  /// in the viewport after it has been revealed. See [RevealedOffset.rect]
+  /// for a full definition of this [Rect].
   ///
   /// The parameters `viewport` and `offset` are required and cannot be null.
-  /// If `child` is null this is a no-op.
-  static void showInViewport({
-    RenderObject child,
+  /// If `descendant` is null this is a no-op and `rect` is returned.
+  ///
+  /// The `duration` parameter can be set to a non-zero value to animate the
+  /// target object into the viewport with an animation defined by `curve`.
+  static Rect showInViewport({
+    RenderObject descendant,
+    Rect rect,
     @required RenderAbstractViewport viewport,
     @required ViewportOffset offset,
+    Duration duration: Duration.zero,
+    Curve curve: Curves.ease,
   }) {
     assert(viewport != null);
     assert(offset != null);
-    if (child == null) {
-      return;
+    if (descendant == null) {
+      return rect;
```

Comment:
should this return `rect ?? viewport.paintBounds` or some such? In the case where rect and descendant are both null, it's not clear returning null is correct, based on the definition of what this returns ("the new location of `descendant` or `rect` in the viewport after it has been revealed").

Reply:
I am documenting that
> If `descendant` is null, this is a no-op and `rect` is returned.

I find that clearer than magically returning the paintBounds of the viewport, which we'd also have to document.

Reply:
what's the use case for both being null?

Reply:
Both being null can happen when you call showOnScreen on the viewport itself. This function would then return null and showOnScreen would be called on the parent with `descendant: viewport and `rect: null`, which would bring the entire Viewport on screen.

We would achieve the same effect by returning viewport.paintBounds here, but like I said, I find that makes reasoning about what showInViewport does harder.